BookTracking.Web/src/pages/Home/BookTable.tsx

```diff
@@ -0,0 +1,119 @@
+import { Fragment, ReactNode, useState } from 'react';
+import { Anchor, Badge, Button, Collapse, Flex, Image, Table } from '@mantine/core';
+import { useDisclosure } from '@mantine/hooks';
+import { Book } from '../../models/Book';
+import { ExpandedBookRow } from './ExpandedBookRow';
+
+import './BookTable.css';
+
+const rowData: Book[] = [
+  {
+    id: '1',
+    authorId: '1',
+    link: new URL('https://www.google.com'),
+    coverImage: './src/assets/testImage.webp',
+    title: 'Book Title',
+    author: 'Author',
+    tags: ['Fantasy', 'Magic'],
+    rating: 7,
+  },
+  {
+    id: '2',
+    authorId: '2',
+    link: new URL('https://www.google.com'),
+    coverImage: './src/assets/testImage.webp',
+    title: 'Book Title 2',
+    author: 'Author',
+    tags: ['Fantasy', 'Magic'],
+    rating: 8,
+  },
+  {
+    id: '3',
+    authorId: '3',
+    link: new URL('https://www.google.com'),
+    coverImage: './src/assets/testImage.webp',
+    title: 'Book Title 3',
+    author: 'Author',
+    tags: ['Fantasy', 'Magic', 'Sci-Fi', 'History', 'Progression', 'Cultivation', 'Fantasy'],
+    rating: 9,
+  },
+  {
+    id: '4',
+    authorId: '4',
+    link: new URL('https://www.google.com'),
+    coverImage: './src/assets/testImage.webp',
+    title: 'Book Title 4',
+    author: 'Author',
+    tags: ['Fantasy', 'Magic'],
+    rating: 10,
+    lastReadBookOrChapter: 'Ch. 23',
+  },
+];
+
+export const BookTable = () => {
+  const [openedRows, setOpenedRows] = useState<number[]>([]);
+
+  const toggleOpened = (index: number) => {
+    setOpenedRows((prev) =>
+      prev.includes(index) ? prev.filter((i) => i !== index) : [...prev, index]
+    );
+  };
+
+  const rows = rowData.map((book, index) => (
+    <>
+      <Table.Tr key={book.id}>
```

Comment:
The `key` prop needs to be on the outermost element in the map for React to correctly optimize updates. In this case put the `key` on the `Fragment`. I suspect there are errors in the browser console mentioning unique key props.